lxd/apparmor/instance.go

```diff
@@ -175,9 +175,13 @@ func instanceProfile(state *state.State, inst instance) (string, error) {
 			}
 		}
 
+		// The value from GetExecPath can include " (deleted)" if the lxd binary has been removed/changed
+		// since the lxd process was started, which can cause the appamor profile load to fail.
+		execPathCleaned := strings.TrimSpace(strings.Replace(util.GetExecPath(), "(deleted)", "", 1))
+
 		err = qemuProfileTpl.Execute(sb, map[string]interface{}{
 			"devPaths":    inst.DevPaths(),
-			"exePath":     util.GetExecPath(),
+			"exePath":     execPathCleaned,
```

Comment:
Does this happen when you actually delete the LXD binary or in other cases as well? And do you only observe this with VMs or with containers?

Reply:
@brauner I can reliably re-create it by deleting the binary (and recreating it), however I originally found the issue after just changing the local lxd and recompiling it and copying it into a VM (working on a separate project) and not restarting the lxd process that was running the VM.  After a few iterations of that I went to stop/start the VM and found I couldn't start it any more due to that error.

I've not seen the issue with containers.

Reply:
Ah, ok. :)

Reply:
@brauner I'm thinking perhaps does `/proc/self/exe` detect when the inode of the file has changed/gone?